### PR TITLE
onlyoffice: drop 7.2.0 version

### PR DESCRIPTION
**Pull request by leon-erd**
## Description of changes

It seems that the versions after `7.2` are now fixed on `wlroots` (see https://github.com/ONLYOFFICE/DesktopEditors/issues/1208#issuecomment-2192315566 and https://github.com/NixOS/nixpkgs/pull/322861). So I guess we don't need the old version anymore.
I think in the file diff it looks a bit messy but the only things I did was delete `onlyoffice-bin`, rename `onlyoffice-bin_latest` to `onlyoffice-bin`, and add information to the aliases. Is there anything else I should do?
Btw why did the `callPackage` statements get removed in this PR https://github.com/NixOS/nixpkgs/pull/285888? I thought every package needs those?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

@nh2 @GTrunSec @fin444

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by fin444 — APPROVED** [review]
Nice to see this split finally removed. Though I'd like to have at least one other person using wlroots review this before merging, just to be safe.

You could have made this part of the update PR if you wanted to save yourself a bit of trouble.

> Btw why did the callPackage statements get removed in this PR https://github.com/NixOS/nixpkgs/pull/285888? I thought every package needs those?

https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md

**Comment by SuperSandro2000** [comment]
> Btw why did the `callPackage` statements get removed in this PR #285888? I thought every package needs those?

Not the packages in pkgs/by-name

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -1025,8 +1025,9 @@ mapAliases ({
   optparse-bash = throw "'optparse-bash' (GitHub: nk412/optparse) has been removed. Use 'argparse' instead"; # Added 2024-01-12
   orchis = orchis-theme; # Added 2021-06-09
   oni2 = throw "oni2 was removed, because it is unmaintained and was abandoned years ago."; #Added 2024-01-15
-  onlyoffice-bin_7_2 = throw "onlyoffice-bin_7_2 has been renamed to onlyoffice-bin"; # Added 2024-02-05
-  onlyoffice-bin_7_5 = throw "onlyoffice-bin_7_5 has been renamed to onlyoffice-bin_latest (and updated from 7.5.x)"; # Added 2024-02-05
+  onlyoffice-bin_7_2 = throw "onlyoffice-bin_7_2 has been renamed to onlyoffice-bin"; # Added 2024-07-03
+  onlyoffice-bin_7_5 = throw "onlyoffice-bin_7_5 has been renamed to onlyoffice-bin"; # Added 2024-07-03
+  onlyoffice-bin_latest = throw "onlyoffice-bin_latest has been renamed to onlyoffice-bin"; # Added 2024-07-03
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:1030]:
```suggestion
  onlyoffice-bin_latest = onlyoffice-bin; # Added 2024-07-03
```

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -1025,8 +1025,9 @@ mapAliases ({
   optparse-bash = throw "'optparse-bash' (GitHub: nk412/optparse) has been removed. Use 'argparse' instead"; # Added 2024-01-12
   orchis = orchis-theme; # Added 2021-06-09
   oni2 = throw "oni2 was removed, because it is unmaintained and was abandoned years ago."; #Added 2024-01-15
-  onlyoffice-bin_7_2 = throw "onlyoffice-bin_7_2 has been renamed to onlyoffice-bin"; # Added 2024-02-05
-  onlyoffice-bin_7_5 = throw "onlyoffice-bin_7_5 has been renamed to onlyoffice-bin_latest (and updated from 7.5.x)"; # Added 2024-02-05
+  onlyoffice-bin_7_2 = throw "onlyoffice-bin_7_2 has been renamed to onlyoffice-bin"; # Added 2024-07-03
+  onlyoffice-bin_7_5 = throw "onlyoffice-bin_7_5 has been renamed to onlyoffice-bin"; # Added 2024-07-03
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:1029]:
```suggestion
  onlyoffice-bin_7_2 = throw "onlyoffice-bin_7_2 has been removed. Please use the latest version available under onlyoffice-bin"; # Added 2024-07-03
  onlyoffice-bin_7_5 = throw "onlyoffice-bin_7_5 has been removed. Please use the latest version available under onlyoffice-bin"; # Added 2024-07-03
```

**Comment by leon-erd** [comment]
> Nice to see this split finally removed. Though I'd like to have at least one other person using wlroots review this before merging, just to be safe.

@fin444 I asked for another confirmation that `8.1.0` works on `wlroots`. Here it is: https://github.com/ONLYOFFICE/DesktopEditors/issues/1208#issuecomment-2248502816
See also https://github.com/ONLYOFFICE/DesktopEditors/issues/1208#issuecomment-2192356570 for another confirmation :)

**Pull request merged**